### DISCUSSION TITLE
How to combine a value and a queue channel

### DISCUSSION THREAD

**Discussion by bounlu** [nextflow-io/nextflow#3934] [Q&A] [answered]
Hi all,

I am trying to combine a string from `params` as a value channel with a queue channel to feed into a process which accepts a tuple of 3 arguments (as part of the `smrnaseq` pipeline).

I tried many things (`.collect()`, `.toList()`, etc.) but they all keep failing for different reasons:

```
params.three_prime_adapter = 'AGATCGGAAGAGCACACGTCT'
trimgalore_reads = FASTQC_TRIMGALORE.out.reads // channel: [ val(meta), [ reads ] ]

Channel.value(params.three_prime_adapter)
    .concat( trimgalore_reads )
    .map { adapterseq, meta, reads -> [adapterseq, meta.id, reads] }
    .groupTuple()
    .set { ch_mirtrace_inputs }

MIRTRACE( ch_mirtrace_inputs )

ERROR ~ Invalid method invocation `call` with arguments: AGATCGGAAGAGCACACGTCT (java.lang.String) on _closure7 type
```

Input for `MIRTRACE` process be like:
```
    input:
    tuple val(adapter), val(ids), path(reads)
```

I need your help to figure out the right way.
Thanks a lot.

**Comment by mribeirodantas** [comment]
The combining part of your code works just fine. The issue is the `map` as your channel has inconsistency between its elements. You have a string for one element and a tuple for the rest. If your channel value was also a tuple `[meta, reads]`, and your `map` closure was referring to two, instead of three items in a single ement, it should work fine. The issue is that no elemnet in your code has three items, just 1 or 2.

**Reply by bounlu** [reply, marked as answer]
You are right. Checking on the console in the light of your explanations helped me to figure it out.

So I just needed to add `.collect()` to make it 3 elements and it worked:

```
    Channel.value(params.three_prime_adapter)
    .concat( trimgalore_reads )
    .collect()
    .map { adapterseq, meta, reads -> [adapterseq, meta.id, reads] }
    .groupTuple()
    .set { ch_mirtrace_inputs }
```
